### ISSUE TITLE
Add option to support updating image on start/restart

### ISSUE THREAD

**Issue by DrewBarrett** [mhutter/ansible-docker-systemd-service#16]
Basically I'm proposing to add a flag that changes the unit file to contain
```
TimeoutStartSec=0
ExecStartPre=-/usr/bin/docker kill image
ExecStartPre=-/usr/bin/docker rm image
ExecStartPre=/usr/bin/docker pull user/image
```
as opposed to the current
```
ExecStartPre=-{{ docker_path }} rm -f {{ container_name }}
```
I'm going to implement this for my own use, so let me know if you'd like me to submit a pull request.

**Comment by mhutter** [comment]
What's the purpose behind this change? The point of `docker rm CONTAINERname` is to avoid conflicts when running `docker run`.

To update the image before (re)starting, only `ExecStartPre=/usr/bin/docker pull user/image` is required.

**Comment by DrewBarrett** [comment]
Yeah that was just an extra verbose example (and `rm image` was a typo for `rm containername`. I believe `TimeoutStartSec=0` is also needed in the case of large images so it does not hit the default systemd timeout
